genesis.bootstrap deboostrap fix

aogier opened the pull request:
### What does this PR do?

It fixes `genesis.bootstrap` deboostrap part by:

 * allowing empty/comma separated strings in `pkgs`/`exclude_pkgs` parameters
 * failing gracefully if no debootstrap command is found in path

### What issues does this PR fix or reference?

closes #43101

Commit:
fix debootstrap and enhance packages selection/deletion via cmdline

salt-jenkins commented:
@aogier, thanks for your PR! By analyzing the history of the files in this pull request, we identified @techhat, @jfindlay and @rallytime to be potential reviewers.

aogier commented:
marking WIP because other enhancements are possible and I'd also branch from correct tag, not develop. Thanks

rallytime commented:
Hi @aogier and welcome! Can you move this change to the 2016.11 branch? The 2016.3 branch is only accepting CVE fixes. Thanks!

aogier commented:
Hi @rallytime, I've rebased my branch on 2016.11 and updated this PR, thank you !

cachedout commented:
Hi @aogier. It looks like you've removed `WIP` from the issue description. Is this now ready to be reviewed?

aogier commented:
yes, thank you !

ale

Il 22 Agosto 2017 17:37:20 CEST, Mike Place <notifications@github.com> ha scritto:
>Hi @aogier. It looks like you've removed `WIP` from the issue
>description. Is this now ready to be reviewed?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/saltstack/salt/pull/43103#issuecomment-324065408

--
Alessandro Ogier
Specialista Reti e Sistemi - Area Ricerca Direzione ICT Gruppo IEO/CCM

** Please note I've changed office, now it
    is 13-3-14 (b. 13, 3rd floor).

Istituto Europeo di Oncologia
Via Serio, 15 - 20139 Milan, Italy
P +39 02 94375269
M +39 334 6841627 (9am-6am CET)
E alessandro.ogier@ieo.eu
W http://www.ieo.it

cachedout commented on salt/modules/genesis.py:
```diff
@@ -393,15 +399,22 @@ def _bootstrap_deb(
+    if pkgs:
+        deb_args += ['--include'] + pkgs
+    if exclude_pkgs:
+        deb_args += ['--exclude'] + exclude_pkgs
```
Should we be wrapping `deb_args` in `cmd_quote()` before we pass it in

aogier replied:
At a first glance I wondered the same thing and I decided it was not necessary but since the relative cost of doing it we probably should opt for the safest path. I'll implement it leaving "exploded" commits for your convenience, I'll squash it later when this PR will hopefully get accepted.

Commit:
allow comma-separated pkgs lists, quote args, test deb behaviour

aogier commented:
Hi, I've just rebased against latest 2016.11. I've implemented quoting and a saner comma-separated pkgs list handling, and wrote unit tests for the debian part.

Thanks, ciao

Commit:
lint

Commit:
tests: fix a leftover and simplify some parts

Commit:
better formatting
